commit afac8c6554ccee54bfd1743755e10af005be3bcf
Author: YiPeng Chai <YiPeng.Chai@amd.com>
Date:   Fri Jul 19 20:43:04 2024 +0800

    drm/amdgpu: fix ras UE error injection failure issue
    
    The ras command shared memory is allocated from
    VRAM and the response status of the command
    buffer will not be zero due to gpu being in
    fatal error state after ras UE error injection.
    
    Signed-off-by: YiPeng Chai <YiPeng.Chai@amd.com>
    Reviewed-by: Hawking Zhang <Hawking.Zhang@amd.com>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    (cherry picked from commit 8284951a6e79c6806c675e5f68a4cd425dd56bc4)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
index 7cdff355cedb..189574d53ebd 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_psp.c
@@ -1630,9 +1630,7 @@ static int psp_ras_send_cmd(struct psp_context *psp,
 
 	switch (cmd) {
 	case TA_RAS_COMMAND__TRIGGER_ERROR:
-		if (ret || psp->cmd_buf_mem->resp.status)
-			ret = -EINVAL;
-		else if (out)
+		if (!ret && out)
 			memcpy(out, &ras_cmd->ras_status, sizeof(ras_cmd->ras_status));
 		break;
 	case TA_RAS_COMMAND__QUERY_ADDRESS: